lib/webapp.js

```diff
@@ -928,6 +929,27 @@ WebServiceHandle.prototype = {
         requestOptions.headers = headers;
       }
 
+      while(this.service && this.service.type === 'import' && serviceModified) {
+        let reqDataservices;
+        serviceModified = false;
+        const reqPlugin = pluginsArray.filter(plugin => plugin.identifier === this.service.sourcePlugin);
+        reqPlugin.filter(plugin => {
+          reqDataservices = plugin.dataServices.filter(dataService => dataService.name === this.service.sourceName);
+        })
+        if(reqDataservices) {
+          const sourcePlugin = this.service.sourcePlugin;
+          const sourceName = this.service.sourceName;
+          this.service = {...reqDataservices[0], sourcePlugin: sourcePlugin, sourceName: sourceName, imported: true };
```

Comment:
If `this.service.sourcePlugin` exists and `this.service.sourceName` exists, shouldn't this just be:
`this.service = {...reqDataservices[0], this.service, imported: true };` or did you intend on removing the other service properties (`service.type` etc)?

Reply:
Yes. I have intended on removing the `service.type` property. `service.type` gets updated depending upon the value in `reqDataservice[0]`